kafka-connect-s3/src/main/java/io/confluent/connect/s3/storage/S3OutputStream.java

```diff
@@ -60,6 +60,8 @@
   private ByteBuffer buffer;
   private MultipartUpload multiPartUpload;
   private final int retries;
+  private final CompressionType compressionType;
+  private OutputStream compressionFilter;
```

Comment:
I'd make this field `volatile` and initialize it the first time `wrapForCompression` is called (instead of init inside the constructor with an unconstructed object - see comment below)